### PR TITLE
Performance adapt

### PR DESCRIPTION
**Pull request by tomasmcm**
Fixes: https://github.com/whitesmith/flooper/issues/20

Instead of targeting mobile devices per se (Mozilla says we shouldn't do it... https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent)
I though it would better to detect performance capability, and limit the animations when the device has a less powerful cpu.

Now, for the JavaScript Magic, apparently there is a property called `navigator.hardwareConcurrency` that returns the number of logical processor cores (https://developer.mozilla.org/en-US/docs/Web/API/NavigatorConcurrentHardware/hardwareConcurrency), it is not supported everywhere though.. (https://caniuse.com/#search=hardwareConcurrency).

So, I found this polyfill that creates WebWorkers to check for performance when `navigator.hardwareConcurrency` is not available https://github.com/phenomnomnominal/workerbench.coffee

The nice thing is because the animations were well implemented by @renatodeleao :kudos:, even if the checking process takes too long (usually no more than 1sec) the website still loads with no animations!

### CODE REVIEW

**Comment by tomasmcm** [comment]
Also, this fixes https://github.com/whitesmith/flooper/issues/17 (it was just a matter of switching the order of the sequence)

File: gulpfile.babel.js [hunk 1]
```diff
@@ -290,11 +290,11 @@ gulp.task("watch", (cb) => {
 
       // WEBPACK HMR did the job with browser sync
       $.watch(`${paths.site.src}/js/**/*`, () => {
-        runSequence("webpack:lib", "webpack:site");
+        runSequence("webpack:site", "webpack:lib");
```
Comment (renatodeleao) [inline gulpfile.babel.js]: Actually now that i'm thinking about it, we don't need to rebuild lib on site change, it has it has nothing to do with it... not sure why I made this :man_facepalming:

Let's remove `runSequence` here

File: gulpfile.babel.js [hunk 2]
```diff
@@ -290,11 +290,11 @@ gulp.task("watch", (cb) => {
...
       });
 
       $.watch(`${paths.lib.src}/**/*`, () => {
-        runSequence("webpack:lib", "webpack:site");
+        runSequence("webpack:site", "webpack:lib");
```
Comment (renatodeleao) [inline gulpfile.babel.js:297]: Really? 🤔  
hmmm... this makes sense as we're now importing the es6 module from the _source_ 
```
import Flooper from '../../lib/index.js',
``` 
the previous though was , w8 for new compiled version of lib => then compile the site, not sure what was wrong. Either way, it's working 🎊 :shipit:

File: site/js/animation.js [hunk 3]
```diff
@@ -1,7 +1,22 @@
 import { indexAnimate } from './animations/indexAnimations.js';
+import 'workerbench.coffee/workerbench.js';
```
Comment (renatodeleao) [inline site/js/animation.js:2]: Never heard about `navigator.hardwareConcurrency`, learnt something today, thanks!

**Review by renatodeleao — COMMENTED** [review]
:shipit:

**Pull request merged**